test/AtlasErrorBubble.t.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "forge-std/Test.sol";
+import { console2 } from "forge-std/console2.sol";
+
+import { BaseTest } from "./base/BaseTest.t.sol";
+import { DummyDAppControl } from "./base/DummyDAppControl.sol";
+import { DummyDAppControlBuilder } from "./helpers/DummyDAppControlBuilder.sol";
+import { CallConfigBuilder } from "./helpers/CallConfigBuilder.sol";
+import { UserOperationBuilder } from "./base/builders/UserOperationBuilder.sol";
+import { SolverOperationBuilder } from "./base/builders/SolverOperationBuilder.sol";
+import { DAppOperationBuilder } from "./base/builders/DAppOperationBuilder.sol";
+
+import { AtlasErrors } from "../src/contracts/types/AtlasErrors.sol";
+import { IDAppControl } from "../src/contracts/interfaces/IDAppControl.sol";
+import { CallConfig } from "../src/contracts/types/ConfigTypes.sol";
+
+import "../src/contracts/types/UserOperation.sol";
+import "../src/contracts/types/SolverOperation.sol";
+import "../src/contracts/types/DAppOperation.sol";
+
+contract NoopSolver {
+    function atlasSolverCall(
+        address,
+        address,
+        address,
+        uint256,
+        bytes calldata,
+        bytes calldata
+    ) external payable { }
+}
+
+contract AtlasErrorBubbleTest is BaseTest {
+
+    DummyDAppControl dAppControl;
+    NoopSolver noopSolver;
+
+    function _deployControl(CallConfig memory callConfig) internal returns (DummyDAppControl) {
+        return new DummyDAppControlBuilder()
+            .withEscrow(address(atlas))
+            .withGovernance(governanceEOA)
+            .withCallConfig(callConfig)
+            .buildAndIntegrate(atlasVerification);
+    }
+
+    function test_bubbles_app_revert_inside_atlas_errors_preOps() public {
```

Comment:
Its a good start but I think we should also add tests for every place we're changing this bubble up behaviour, not just preOps. Especially tests that show the dapp control, userOp call, or solver calls cannot cause some malicious data to bubble up, causing us to treat the error in a way it shouldn't be treated